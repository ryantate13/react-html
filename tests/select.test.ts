import {select} from '../src';

describe('select', () => {
    it('renders with children', () => {
        expect(select({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(select({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(select({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(select()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(select({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(select({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(select({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(select({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(select({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(select({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(select({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(select({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(select({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(select({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(select({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(select({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(select({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(select({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(select({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(select({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(select({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(select({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(select({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(select({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(select({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(select({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(select({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(select({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(select({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(select({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(select({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(select({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(select({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(select({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(select({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(select({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(select({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(select({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(select({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(select({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(select({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(select({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(select({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(select({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(select({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(select({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(select({translate: "no"})).toMatchSnapshot();
    });

    it('renders with autoComplete="off"', () => {
        expect(select({autoComplete: "off"})).toMatchSnapshot();
    });

    it('renders with disabled=true', () => {
        expect(select({disabled: true})).toMatchSnapshot();
    });

    it('renders with disabled=false', () => {
        expect(select({disabled: false})).toMatchSnapshot();
    });

    it('renders with form="TEST"', () => {
        expect(select({form: "TEST"})).toMatchSnapshot();
    });

    it('renders with multiple=true', () => {
        expect(select({multiple: true})).toMatchSnapshot();
    });

    it('renders with multiple=false', () => {
        expect(select({multiple: false})).toMatchSnapshot();
    });

    it('renders with name="TEST"', () => {
        expect(select({name: "TEST"})).toMatchSnapshot();
    });

    it('renders with required=true', () => {
        expect(select({required: true})).toMatchSnapshot();
    });

    it('renders with required=false', () => {
        expect(select({required: false})).toMatchSnapshot();
    });

    it('renders with size=1', () => {
        expect(select({size: 1})).toMatchSnapshot();
    });
});