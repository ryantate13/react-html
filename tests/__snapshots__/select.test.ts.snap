// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select renders arrays of children 1`] = `
<select>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</select>
`;

exports[`select renders with accessKey="a,b,c,d" 1`] = `
<select
  accessKey="a,b,c,d"
/>
`;

exports[`select renders with autoCapitalize="characters" 1`] = `
<select
  autoCapitalize="characters"
/>
`;

exports[`select renders with autoCapitalize="none" 1`] = `
<select
  autoCapitalize="none"
/>
`;

exports[`select renders with autoCapitalize="off" 1`] = `
<select
  autoCapitalize="off"
/>
`;

exports[`select renders with autoCapitalize="on" 1`] = `
<select
  autoCapitalize="on"
/>
`;

exports[`select renders with autoCapitalize="sentences" 1`] = `
<select
  autoCapitalize="sentences"
/>
`;

exports[`select renders with autoCapitalize="words" 1`] = `
<select
  autoCapitalize="words"
/>
`;

exports[`select renders with autoComplete="off" 1`] = `
<select
  autoComplete="off"
/>
`;

exports[`select renders with children 1`] = `
<select>
  TEST
</select>
`;

exports[`select renders with className="foo bar baz" 1`] = `
<select
  className="foo bar baz"
/>
`;

exports[`select renders with contentEditable="false" 1`] = `
<select
  contentEditable="false"
/>
`;

exports[`select renders with contentEditable="true" 1`] = `
<select
  contentEditable="true"
/>
`;

exports[`select renders with dir="auto" 1`] = `
<select
  dir="auto"
/>
`;

exports[`select renders with dir="ltr" 1`] = `
<select
  dir="ltr"
/>
`;

exports[`select renders with dir="rtl" 1`] = `
<select
  dir="rtl"
/>
`;

exports[`select renders with disabled=false 1`] = `
<select
  disabled={false}
/>
`;

exports[`select renders with disabled=true 1`] = `
<select
  disabled={true}
/>
`;

exports[`select renders with draggable="false" 1`] = `
<select
  draggable="false"
/>
`;

exports[`select renders with draggable="true" 1`] = `
<select
  draggable="true"
/>
`;

exports[`select renders with empty attributes 1`] = `<select />`;

exports[`select renders with form="TEST" 1`] = `
<select
  form="TEST"
/>
`;

exports[`select renders with hidden=false 1`] = `
<select
  hidden={false}
/>
`;

exports[`select renders with hidden=true 1`] = `
<select
  hidden={true}
/>
`;

exports[`select renders with id="TEST" 1`] = `
<select
  id="TEST"
/>
`;

exports[`select renders with inputMode="decimal" 1`] = `
<select
  inputMode="decimal"
/>
`;

exports[`select renders with inputMode="email" 1`] = `
<select
  inputMode="email"
/>
`;

exports[`select renders with inputMode="none" 1`] = `
<select
  inputMode="none"
/>
`;

exports[`select renders with inputMode="numeric" 1`] = `
<select
  inputMode="numeric"
/>
`;

exports[`select renders with inputMode="search" 1`] = `
<select
  inputMode="search"
/>
`;

exports[`select renders with inputMode="tel" 1`] = `
<select
  inputMode="tel"
/>
`;

exports[`select renders with inputMode="text" 1`] = `
<select
  inputMode="text"
/>
`;

exports[`select renders with inputMode="url" 1`] = `
<select
  inputMode="url"
/>
`;

exports[`select renders with is="foo" 1`] = `
<select
  is="foo"
/>
`;

exports[`select renders with itemID="https://google.com/" 1`] = `
<select
  itemID="https://google.com/"
/>
`;

exports[`select renders with itemProp="https://google.com/" 1`] = `
<select
  itemProp="https://google.com/"
/>
`;

exports[`select renders with itemRef="foo bar" 1`] = `
<select
  itemRef="foo bar"
/>
`;

exports[`select renders with itemScope=false 1`] = `
<select
  itemScope={false}
/>
`;

exports[`select renders with itemScope=true 1`] = `
<select
  itemScope={true}
/>
`;

exports[`select renders with itemType="https://google.com/" 1`] = `
<select
  itemType="https://google.com/"
/>
`;

exports[`select renders with lang="" 1`] = `
<select
  lang=""
/>
`;

exports[`select renders with lang="es-419" 1`] = `
<select
  lang="es-419"
/>
`;

exports[`select renders with multiple=false 1`] = `
<select
  multiple={false}
/>
`;

exports[`select renders with multiple=true 1`] = `
<select
  multiple={true}
/>
`;

exports[`select renders with name="TEST" 1`] = `
<select
  name="TEST"
/>
`;

exports[`select renders with required=false 1`] = `
<select
  required={false}
/>
`;

exports[`select renders with required=true 1`] = `
<select
  required={true}
/>
`;

exports[`select renders with size=1 1`] = `
<select
  size={1}
/>
`;

exports[`select renders with slot="TEST" 1`] = `
<select
  slot="TEST"
/>
`;

exports[`select renders with spellCheck="false" 1`] = `
<select
  spellCheck="false"
/>
`;

exports[`select renders with spellCheck="true" 1`] = `
<select
  spellCheck="true"
/>
`;

exports[`select renders with style={"color":"#000000"} 1`] = `
<select
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`select renders with tabIndex=1 1`] = `
<select
  tabIndex={1}
/>
`;

exports[`select renders with title="TEST" 1`] = `
<select
  title="TEST"
/>
`;

exports[`select renders with translate="no" 1`] = `
<select
  translate="no"
/>
`;

exports[`select renders with translate="yes" 1`] = `
<select
  translate="yes"
/>
`;

exports[`select renders without attributes 1`] = `<select />`;
